app/addons/documents/rightalldocsheader.react.jsx

```diff
@@ -0,0 +1,57 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  "../../app",
+  "../../core/api",
+  "../fauxton/components",
+  "./resources",
+  "../databases/resources",
+  "React",
+
+  // Views
+  "./queryoptions/queryoptions.react",
+  "./queryoptions/actions",
+  './jumptodoc.react',
+  './queryoptions/stores',
+
+  //plugins
+  "../../../assets/js/plugins/prettify"
+], function (app, FauxtonAPI, Components, Documents, Databases, React, {QueryOptionsController}, QueryActions, {JumpToDoc},
+  { queryOptionsStore }) {
+
+
+  const RightAllDocsHeader = React.createClass({
```

Comment:
general question, just curious: any reason you don't use ES6 classes (`class RightAllDocsHeader extends React.Component {};`?